docarray/array/doc_list/doc_list.py

```diff
@@ -337,8 +337,11 @@ def __class_getitem__(cls, item: Union[Type[BaseDoc], TypeVar, str]):
 
         if isinstance(item, type) and safe_issubclass(item, BaseDoc):
             return AnyDocArray.__class_getitem__.__func__(cls, item)  # type: ignore
-        else:
-            return super().__class_getitem__(item)
+
+        if isinstance(item, object) and not isinstance(item, TypeVar):
```

Comment:
added a check `isinstance(item, TypeVar)` since item of `TypeVar` was also being passed when `__class_getitem__` was called